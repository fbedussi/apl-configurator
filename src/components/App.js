import React from 'react';
import { connect } from 'react-redux';
import ReactCSSTransitionGroup from 'react-addons-css-transition-group';

import { getSteps, getStepsLeft, getCurrentNode, getBreadcrumbs, getLabels, getGoingBack, getAnswers } from '../selectors';
import { parseAnswer, goBack } from '../actions';

import Answers from './Answers';
import Title from './Title';
import Back from './Back';
import Image from './Image';
import Recap from './Recap';
import Quotation from './Quotation';

const mapStateToProps = (state) => ({
    labels: getLabels(state),
    steps: getSteps(state),
    stepsLeft: getStepsLeft(state),
    goingBack: getGoingBack(state),
    currentNode: getCurrentNode(state),
    breadcrumbs: getBreadcrumbs(state),
    answers: getAnswers(state)
});

const mapDispatchToProps = (dispatch) => ({
    setAnswer: (value) => dispatch(parseAnswer(value)),
    goBack: (currentNode) => dispatch(goBack(currentNode))
})

class App extends React.Component {
    render() {
        return (
            <div className="configuratorWrapper">
                <div className="progressWrapper">
                    <span className="progressLabel">{this.props.labels["configurator"]}</span>
                    <progress className="progressBar" value={this.props.steps - this.props.stepsLeft} max={this.props.steps}></progress>
                </div>
                <div className={`workarea ${this.props.goingBack? 'back' : ''}`}>
                    <ReactCSSTransitionGroup
                        transitionName="slider"
                        transitionEnterTimeout={500}
                        transitionLeaveTimeout={500}>
                        <div className="slide" key={this.props.currentNode.id}>
                            <Title currentNode={this.props.currentNode} />
                            <p className={this.props.currentNode.type}>{this.props.currentNode.text}</p>
                            <Image 
                                imgName={this.props.currentNode.image}
                            />
                            <Answers
                                id={this.props.currentNode.id}
                                answers={this.props.currentNode.answers}
                                setAnswer={this.props.setAnswer}
                            />
                            <Back
                                show={Boolean(this.props.breadcrumbs.length)}
                                label={this.props.labels["back"]}
                                clickHandler={() =>
                                    this.props.goBack(this.props.currentNode)
                                }
                            />
                            <Recap
                                show={this.props.currentNode.type === 'answer'}
                                breadcrumbs={this.props.breadcrumbs}
                                answers={this.props.answers}
                            />
                            <Quotation
                                show={this.props.currentNode.type === 'answer'}
                            />
                        </div>

                    </ReactCSSTransitionGroup>

                </div>
            </div>
        );
    }
};

export default connect(mapStateToProps, mapDispatchToProps)(App);