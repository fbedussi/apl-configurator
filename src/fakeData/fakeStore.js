import texts from './apl-texts.json';

const fakeStore = {
    currentLanguage: texts.languageIsoCode,
    showForm: false,
    steps: 6,
    stepsLeft: 6,
    goingBack: false,
    breadcrumbs: [],
    answersHistory: [],
    labels: texts.labels,
    answers: texts.answers,
    questions: texts.questions,
    tree: texts.tree,
    currentNode: texts.tree,
    requestStatus: 'unsent' // || sent || success || error
};

export default fakeStore;
