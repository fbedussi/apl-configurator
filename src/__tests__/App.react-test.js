import React from 'react';
import { createStore } from 'redux';
import { Provider } from 'react-redux';

import App from '../components/App';
import renderer from 'react-test-renderer';
import fakeStore from '../fakeData/fakeStore';

const store = createStore(() => fakeStore);

test('App component', () => {
  const component = renderer.create(
    <Provider store={store}>
      <App />
    </Provider>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
