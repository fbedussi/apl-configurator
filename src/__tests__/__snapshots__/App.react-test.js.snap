// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component 1`] = `
<div
  className="appInner"
  lang="it"
>
  <header
    className="header"
  >
    <div
      className="wrapper header-wrapper"
    >
      <a
        className="logoLink"
        href="http://www.attraversamentipedonali.it/"
      >
        <img
          alt="torna al sito attraversamentipedonali.it"
          src="images/logo_apl-02.png"
        />
      </a>
      <a
        className="linkHome"
        href="http://www.attraversamentipedonali.it/"
      >
        Home page &gt;
      </a>
    </div>
  </header>
  <div
    className="wrapper main-wrapper"
  >
    <div
      className="progressWrapper"
    >
      <span
        className="progressLabel"
      >
        Configuratore
      </span>
      <progress
        className="progressBar"
        max={6}
        value={0}
      />
    </div>
    <div
      className="workarea "
    >
      <div
        className="slide"
      >
        <div
          className="slideInner question"
        >
          <p
            className="text question-text"
            dangerouslySetInnerHTML={
              Object {
                "__html": "Nel luogo di installazione è possibile collegare il sistema APL alla rete elettrica?",
              }
            }
          />
          <div
            className="question-options"
          >
            <button
              className="question-optionBtn ctaBtn"
              id="1_0"
              onClick={[Function]}
            >
              no
            </button>
            <button
              className="question-optionBtn ctaBtn"
              id="1_1"
              onClick={[Function]}
            >
              sì
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
  <footer
    className="footer"
  >
    <div
      className="wrapper footer-wrapper"
    >
      <div
        className="footer-logoAndLink"
      >
        <img
          alt={undefined}
          src="images/d-power-logo_70.png"
        />
        <a
          className="linkHome"
          href="http://www.attraversamentipedonali.it/"
        >
          Home page &gt;
        </a>
      </div>
      <small
        className="copyright"
      >
        © Detas.com - All rights reserved
      </small>
    </div>
  </footer>
</div>
`;
